src/background/services/openPayments.ts

```diff
@@ -881,6 +884,59 @@ export class OpenPaymentsService {
     return outgoingPayment;
   }
 
+  /** Polls for the completion of an outgoing payment */
+  async *pollOutgoingPayment(
+    outgoingPaymentId: OutgoingPayment['id'],
+    {
+      signal,
+      maxAttempts = 10,
+    }: Partial<{ signal: AbortSignal; maxAttempts: number }> = {},
+  ): AsyncGenerator<OutgoingPayment, OutgoingPayment, void> {
+    let attempt = 0;
+    let tokenRotated = false;
+    await sleep(OUTGOING_PAYMENT_POLLING_INITIAL_DELAY);
+    while (++attempt <= maxAttempts) {
+      try {
+        signal?.throwIfAborted();
+        const outgoingPayment = await this.client!.outgoingPayment.get({
+          url: outgoingPaymentId,
+          accessToken: this.token.value,
+        });
+        yield outgoingPayment;
+        if (
+          outgoingPayment.failed &&
+          outgoingPayment.sentAmount.value === '0'
+        ) {
+          throw new ErrorWithKey('pay_error_outgoingPaymentFailed');
+        }
+        if (
+          outgoingPayment.debitAmount.value === outgoingPayment.sentAmount.value
+        ) {
+          return outgoingPayment; // completed
+        }
+        signal?.throwIfAborted();
+        await sleep(OUTGOING_PAYMENT_POLLING_INTERVAL);
+      } catch (error) {
+        if (isMissingGrantPermissionsError(error)) {
+          if (isTokenInactiveError(error) && !tokenRotated) {
+            // isMissingGrantPermissionError has same error msg as isTokenInactiveError
+            tokenRotated = true;
+            await this.rotateToken();
+            continue;
+          }
+          throw error;
+        } else if (isTokenExpiredError(error) && !tokenRotated) {
+          tokenRotated = true;
+          await this.rotateToken();
```

Comment:
Maybe a better idea would be check response of `rotateToken` to see if token as required permission. Will do.